LochNessBuilder/Builder.cs

```diff
@@ -142,8 +142,18 @@ public Builder<TInstance> WithFactory<TProp>(Expression<Func<TInstance, TProp>>
             var instance = GetInstance();
             var prop = GetProp(selector, instance);
             Expression<Func<TProp>> valueInvoker = () => value();
+            Expression setExpression;
```

Comment:
I personally avoid initializing a variable without assignment where I can - instead, you could have a "createSetExpressionForValue<TProp>(property, valueGetter)" or something, which has a couple of return statements

Reply:
Got refactored.